observability-lib/grafana/builder.go

```diff
@@ -104,33 +111,39 @@ func (b *Builder) AddNotificationPolicy(notificationPolicies ...*alerting.Notifi
 	b.notificationPoliciesBuilder = append(b.notificationPoliciesBuilder, notificationPolicies...)
 }
 
-func (b *Builder) Build() (*Dashboard, error) {
-	db, errBuildDashboard := b.dashboardBuilder.Build()
-	if errBuildDashboard != nil {
-		return nil, errBuildDashboard
-	}
+func (b *Builder) Build() (*Observability, error) {
+	observability := Observability{}
 
-	var alerts []alerting.Rule
-	for _, alertBuilder := range b.alertsBuilder {
-		alert, errBuildAlert := alertBuilder.Build()
-		if errBuildAlert != nil {
-			return nil, errBuildAlert
+	if b.dashboardBuilder != nil {
+		db, errBuildDashboard := b.dashboardBuilder.Build()
+		if errBuildDashboard != nil {
+			return nil, errBuildDashboard
```

Comment:
/nit, but when passing errors up the stack, it can be helpful for debugging to wrap the error with a `fmt.Errorf` with a description of what the consumer observed with that error, i.e. 

```
return nil, fmt.Errorf("failure in building dashboard: %w, errBuildDashboard")
```

Reply:
right I'll take care of this in future iterations